eventuals/generator.h

```diff
@@ -436,16 +520,26 @@ struct _Generator final {
                 std::move(body),
                 std::move(ended));
             break;
-          case Action::Fail:
-            e->Fail(
-                interrupt,
-                std::move(exception.value()),
-                std::move(begin),
-                std::move(fail),
-                std::move(stop),
-                std::move(body),
-                std::move(ended));
-            break;
+          case Action::Fail: {
+            // If 'Catches_' is empty then we will never dispatch
+            // with an action of 'Action::Fail' but since 'action' is a runtime
+            // value the compiler will assume that it's possible that we can
+            // call 'e->Fail()', with what ever type we have for 'error' so to
+            // keep the compiler from trying to compile that code path we need
+            // to add the following 'if constexpr'.
+            if constexpr (std::tuple_size_v<Catches_> != 0) {
```

Comment:
I don't understand why we need both this `if constexpr` and the one in `HeapGenerator::Fail`. Seems like we should only need one of them!

Reply:
We need two of them: one is in Fail to be sure we invoke dispatch with Catch error, second is inside Dispatch (prevents compiler to go in fail path with a wrong error)